pkg/query/querier.go

```diff
@@ -268,12 +286,17 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 
 	// TODO(bwplotka): Use inprocess gRPC.
 	resp := &seriesServer{ctx: ctx}
+	var queryHints storepb.QueryHints
+	if q.enableQueryPushdown {
+		queryHints = storeHintsFromPromHints(hints)
+	}
 	if err := q.proxy.Series(&storepb.SeriesRequest{
 		MinTime:                 hints.Start,
 		MaxTime:                 hints.End,
 		Matchers:                sms,
 		MaxResolutionWindow:     q.maxResolutionMillis,
 		Aggregates:              aggrs,
+		QueryHints:              &queryHints,
```

Comment:
But now `QueryHints` will always be not nil, right? I think there's a bug now here. It should be something like:

```go
var queryHints *storepb.QueryHints
if q.enableQueryPushdown {
  generatedHints := storeHintsFromPromHints(hints)
  queryHints = generatedHints
}
```

@fpetkovski :thinking:

Reply:
Yes, you're right, good catch. I applied your suggestion